console/src/components/Login/Login.js

```diff
@@ -1,96 +1,115 @@
 import globals from '../../Globals';
-import { loginClicked, UPDATE_ADMIN_SECRET_INPUT } from '../Main/Actions';
+import { verifyLogin } from './Actions';
 
-class Login extends Component {
-  handleAdminSecret = e => {
-    this.props.dispatch({
-      type: UPDATE_ADMIN_SECRET_INPUT,
-      data: e.target.value,
-    });
-  };
+const styles = require('./Login.scss');
+const hasuraLogo = require('./blue-logo.svg');
 
-  loginClicked = () => {
-    this.props.dispatch(loginClicked());
-  };
+const Login = ({ dispatch }) => {
+  // request state
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
 
-  render() {
-    const { loginInProgress, loginError, dispatch } = this.props;
+  // should persist admin secret
+  const [shouldPersist, setShouldPersist] = useState(true);
+  const toggleShouldPersist = () => setShouldPersist(!shouldPersist);
 
-    const styles = require('./Login.scss');
-    const hasuraLogo = require('./blue-logo.svg');
+  // input handler
+  const [adminSecretInput, setAdminSecretInput] = useState('');
+  const onAdminSecretChange = e => setAdminSecretInput(e.target.value);
 
-    let loginText = 'Enter';
-    if (loginInProgress) {
-      loginText = (
-        <span>
-          Verifying...
-          <i className="fa fa-spinner fa-spin" aria-hidden="true" />
-        </span>
-      );
-    } else if (loginError) {
-      loginText = 'Error. Try again?';
-    }
+  // login button text
+  let loginText = 'Enter';
+  if (loading) {
+    loginText = (
+      <span>
+        Verifying...
+        <i className="fa fa-spinner fa-spin" aria-hidden="true" />
+      </span>
+    );
+  } else if (error) {
+    loginText = 'Error. Try again?';
+  }
 
-    return (
-      <div className={styles.mainWrapper + ' container-fluid'}>
-        <div className={styles.container + ' container'} id="login">
-          <div className={styles.loginCenter}>
-            <Helmet title={'Login | ' + 'Hasura'} />
-            <div className={styles.hasuraLogo}>
-              <img src={hasuraLogo} />
-            </div>
-            <div className={styles.loginWrapper}>
-              <form
-                className="form-horizontal"
-                onSubmit={e => {
-                  e.preventDefault();
-                  dispatch(loginClicked());
-                }}
-              >
-                <div
-                  className={styles.input_addon_group + ' ' + styles.padd_top}
-                >
-                  <div className={'input-group ' + styles.input_group}>
-                    <input
-                      onChange={this.handleAdminSecret}
-                      className={styles.form_input + ' form-control'}
-                      type="password"
-                      placeholder={`Enter ${globals.adminSecretLabel}`}
-                      name="password"
-                    />
-                  </div>
+  // form submit handler
+  const onSubmit = e => {
```

Comment:
Cannot read this code block. There are 5 distinct things happening here which I need to read line by line to figure out. Please add spacing after logical blocks

```
 const onSubmit = e => {
    e.preventDefault();

    const successCallback = () => {
      setLoading(false);
      setError(null);
    };

    const errorCallback = err => {
      setAdminSecretInput('');
      setLoading(false);
      setError(err);
    };

    setLoading(true);

    verifyLogin({
      adminSecret: adminSecretInput,
      shouldPersist,
      successCallback,
      errorCallback,
      dispatch,
    });
  };
```